src/agent/rest_web_server.cpp

```diff
@@ -0,0 +1,1061 @@
+#include <netinet/in.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <chrono>
+#include <algorithm> 
+#include <iostream> 
+#include <fstream> 
+#include <string>
+#include <sstream>
+#include <errno.h>
+#include <sys/time.h>
+#include <string>
+#include "openthread/thread_ftd.h"
+#include "common/logging.hpp"
+
+#include "agent/rest_web_server.hpp"
+
+#include "cJSON.h"
+#include "cJSON.c"
+#include "http_parser.h"
+#include "http_parser.c"
+
+
+
+namespace otbr {
+namespace agent {
+
+
+static int on_message_begin(http_parser *parser) {
```

Comment:
Follow OpenThread naming style:
```suggestion
static int OnMessageBegin(http_parser *parser) {
```